src/shared/extensions/ssh.ts

```diff
@@ -123,9 +123,10 @@ export async function startVscodeRemote(
     ProcessClass: typeof ChildProcess,
     hostname: string,
     targetDirectory: string,
-    vscPath: string
+    vscPath: string,
+    user?: string
 ): Promise<void> {
-    const workspaceUri = `vscode-remote://ssh-remote+${hostname}${targetDirectory}`
+    const workspaceUri = `vscode-remote://ssh-remote+${`${user}@` ?? ''}${hostname}${targetDirectory}`
```

Comment:
The `??` in `${user}@ ?? ''` never happens because of the "@". The string is getting complicated so better treat the user can separately

```suggestion
    const userAt = user ? `${user}@` : ''
    const workspaceUri = `vscode-remote://ssh-remote+${userAt}${hostname}${targetDirectory}`
```